src/include/columnar/columnar_tableam.h

```diff
@@ -29,7 +29,7 @@
  * For this reason, we restrict itemPointer.offsetNumber
  * to the following interval: [FirstOffsetNumber, MaxOffsetNumber).
  */
-#define VALID_ITEMPOINTER_OFFSETS (MaxOffsetNumber - FirstOffsetNumber)
+#define VALID_ITEMPOINTER_OFFSETS ((uint64) (MaxOffsetNumber - FirstOffsetNumber))
```

Comment:
MaxHeapTuplesPerPage for now.

Reply:
Actually I made that change already, but not sure why github shows like that:
https://github.com/citusdata/citus/pull/4958#discussion_r628248431